fastly/base_fastly_service_v1.go

```diff
@@ -0,0 +1,482 @@
+	GetAttributeHandler() []ServiceAttributeDefinition
+}
+
+// BaseServiceDefinition is the base implementation of the BaseServiceDefinition interface
+type BaseServiceDefinition struct {
+	Attributes []ServiceAttributeDefinition
+	Type       string
+}
+
+func (d *BaseServiceDefinition) GetType() string {
+	return d.Type
+}
+
+func (d *BaseServiceDefinition) GetAttributeHandler() []ServiceAttributeDefinition {
+	return d.Attributes
+}
+
+// resourceService returns a Terraform resource schema for VCL or WASM
+func resourceService(serviceDef ServiceDefinition) *schema.Resource {
+	s := &schema.Resource{
+		Create: resourceCreate(serviceDef),
+		Read:   resourceRead(serviceDef),
+		Update: resourceUpdate(serviceDef),
+		Delete: resourceDelete(serviceDef),
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Unique name for this Service",
+			},
+
+			"comment": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "Managed by Terraform",
+				Description: "A personal freeform descriptive note",
+			},
+
+			"version_comment": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A personal freeform descriptive note",
+			},
+
+			// Active Version represents the currently activated version in Fastly. In
+			// Terraform, we abstract this number away from the users and manage
+			// creating and activating. It's used internally, but also exported for
+			// users to see.
+			"active_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			// Cloned Version represents the latest cloned version by the provider. It
+			// gets set whenever Terraform detects changes and clones the currently
+			// activated version in order to modify it. Active Version and Cloned
+			// Version can be different if the Activate field is set to false in order
+			// to prevent the service from being activated. It is not used internally,
+			// but it is exported for users to see after running `terraform apply`.
+			"cloned_version": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"activate": {
+				Type:        schema.TypeBool,
+				Description: "Conditionally prevents the Service from being activated",
+				Default:     true,
+				Optional:    true,
+			},
+
+			"force_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+		},
+	}
+
+	for _, a := range serviceDef.GetAttributeHandler() {
+		a.Register(s) // Mutates s, adding handler-specific schema items to the list
+	}
+
+	return s
+}
+
+// resourceCreate satisfies the Terraform resource schema Create "interface"
+// while injecting the ServiceDefinition into the true Create functionality
+func resourceCreate(serviceDef ServiceDefinition) schema.CreateFunc {
+	return func(data *schema.ResourceData, i interface{}) error {
```

Comment:
Suggestion:

It might be a little clearer if we propagate the `meta` param name to this level.
```suggestion
	return func(data *schema.ResourceData, meta interface{}) error {
```